### PR TITLE
controller: made QPS, burst QPS and number of workers externally configurable

### PR DESCRIPTION
**Pull request by jkowalski**
Also increased their default values significantly based on scale test results.

### CODE REVIEW

**Comment by markmandel** [comment]
Do we want to expose these as helm variables? or wait and see for now if people want to adjust them?

**Comment by jkowalski** [comment]
Let me add helm config...

**Comment by markmandel** [comment]
This looks great, but you got caught by the new website docs causing conflicts for you. Sorry!

File: site/content/en/docs/Installation/helm.md [hunk 1]
```diff
@@ -129,14 +129,17 @@ The following tables lists the configurable parameters of the Agones chart and t
 | `agones.controller.healthCheck.timeoutSeconds`      | Number of seconds after which the probe times out (in seconds)                                  | `1`                    |
 | `agones.controller.resources`                       | Controller resource requests/limit                                                              | `{}`                   |
 | `agones.controller.generateTLS`                     | Set to true to generate TLS certificates or false to provide your own certificates in `certs/*` | `true`                 |
+| `agones.controller.numWorkers`                      | Number of workers to spin per resource type                                                     | `64`                   |
```
Comment (markmandel) [inline site/content/en/docs/Installation/helm.md]: This section should be below in the "new configuration features" section. (best way I could handle it in the docs)

Comment (jkowalski) [reply site/content/en/docs/Installation/helm.md]: done

File: test/e2e/gameserver_test.go [hunk 2]
```diff
@@ -109,13 +108,12 @@ func TestSDKSetAnnotation(t *testing.T) {
...
 			return true, err
 		}
-		return gs.ObjectMeta.Annotations != nil, nil
+
+		// TODO - due to #455 the version sometimes disappears, add check when it's stable
+		return gs.ObjectMeta.Annotations["stable.agones.dev/sdk-timestamp"] != "", nil
 	})
 
 	assert.Nil(t, err)
-	assert.NotEmpty(t, gs.ObjectMeta.Annotations["stable.agones.dev/sdk-timestamp"])
```
Comment (markmandel) [inline test/e2e/gameserver_test.go]: This is now fixed in #516

Comment (markmandel) [reply test/e2e/gameserver_test.go]: Now that #516 is fixed, let's remove this change please.

File: site/content/en/docs/Installation/helm.md [hunk 3]
```diff
@@ -130,13 +130,13 @@ The following tables lists the configurable parameters of the Agones chart and t
...
 | `agones.controller.generateTLS`                     | Set to true to generate TLS certificates or false to provide your own certificates in `certs/*` | `true`                 |
 | `agones.ping.install`                               | Whether to install the [ping service][ping]                                                     | `true`                 |
-| `agones.ping.replicas`                              | The number of replicas to run in the deployment                                                 | `2`                    | 
-| `agones.ping.http.expose`                           | Expose the http ping service via a Service                                                      | `true`                 | 
-| `agones.ping.http.response`                         | The string response returned from the http service                                              | `ok`                   | 
+| `agones.ping.replicas`                              | The number of replicas to run in the deployment                                                 | `2`                    |
```
Comment (markmandel) [inline site/content/en/docs/Installation/helm.md]: These new configuration params need to be moved to the "new configuration features" section below as well, so they are hidden when I work on CD for the site this week.

Comment (jkowalski) [reply site/content/en/docs/Installation/helm.md]: those *ping* changes are not part of my change. I think we should move them in separate commit.

Comment (markmandel) [reply site/content/en/docs/Installation/helm.md]: ignore me. they are trailing line removals.

**Pull request merged**